catalog/app/components/Preview/renderers/Perspective.js

```diff
@@ -0,0 +1,102 @@
+import cx from 'classnames'
+import * as React from 'react'
+import * as M from '@material-ui/core'
+
+import * as perspective from 'utils/perspective'
+import { readableBytes } from 'utils/string'
+
+import { CONTEXT } from '../types'
+
+function SizeEstimation({ current, full }) {
+  if (!current || !full) return null
+  return (
+    <span>
+      (est. {readableBytes(current)} of {readableBytes(full)})
+    </span>
+  )
+}
+
+const useTruncatedWarningStyles = M.makeStyles((t) => ({
+  root: {
+    alignItems: 'center',
+    display: 'flex',
+  },
+  message: {
+    color: t.palette.text.secondary,
+    marginRight: t.spacing(2),
+  },
+  icon: {
+    display: 'inline-block',
+    fontSize: '1.25rem',
+    marginRight: t.spacing(0.5),
+    verticalAlign: '-5px',
+  },
+}))
+
+function TruncatedWarning({ className, onLoadMore }) {
+  const classes = useTruncatedWarningStyles()
+  return (
+    <div className={cx(classes.root, className)}>
+      <span className={classes.message}>
+        <M.Icon fontSize="small" color="inherit" className={classes.icon}>
+          info_outlined
+        </M.Icon>
+        Partial preview <SizeEstimation />
```

Comment:
props aren't passed to `SizeEstimation`, therefore it always returns null and renderes nothing. is this intentional?